metric2/asyncmetric/asyncmetric.go

```diff
@@ -0,0 +1,27 @@
+package asyncmetric
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/metric2/instrument"
+	"go.opentelemetry.io/otel/metric2/asyncmetric/asyncfloat64metric"
+	"go.opentelemetry.io/otel/metric2/asyncmetric/asyncint64metric"
+)
+
+type Meter struct {
+}
+
+type Callback struct {
```

Comment:
Thanks for pointing this out. The Specification is moving to use "Asynchronous" to describe the instruments that use a callback. Yes, this object is a Handler, but the Spec also uses "Callback" to describe the active piece of code that makes observations. We can call it "Handler" in Go if everyone agrees.

I was trying to steer this API toward a lower-level interface than the former API. This Callback is the function-object passed to the BatchObserver instrument in the former API, here: https://github.com/open-telemetry/opentelemetry-go/blob/484258eb36a71e7495709da0a85c0b12d6d39176/metric/metric.go#L54

and the idea is to only support this batch-oriented API for async instruments. Helpers could be provided for singleton observers, and of course this would be way easier with generics.